### ISSUE TITLE
Generating Multiple Masks with multiple prompt points

### ISSUE THREAD

**Issue by ehdmso** [facebookresearch/segment-anything#531]
Referring to the content in [predictor_example.ipynb](https://github.com/facebookresearch/segment-anything/blob/main/notebooks/predictor_example.ipynb),  the concept of "Batched prompt inputs" is introduced through the utilization of `transformed_boxes.`

I am seeking guidance on the process of generating multiple masks to accommodate various input points. To illustrate, I intend to create two or more distinct masks, with each mask requiring three or more specific prompt points.

Could you provide assistance on how to handle this scenario?

---
what i have tried : 

1. The information suggests referring to the module labeled as `segment_anything.utils.transforms`.

 https://github.com/facebookresearch/segment-anything/blob/6fdee8f2727f4506cfbbe553e23b895e27956588/segment_anything/utils/transforms.py#L67

2. I tried `apply_coords_torch`

![Untitled (1)](https://github.com/facebookresearch/segment-anything/assets/114130205/5ef5a9a8-ed4c-46fe-94ac-9bbedb4a0d69)

However, there is some confusion concerning whether `input_points` should be implemented as a torch.tensor or a numpy array.

**Comment by heyoeyo** [comment]
As far as I can tell, your `input_points` should be a tensor, similar to the `input_boxes` example in the notebook. The `input_labels` should also be a tensor. I believe you also need to use the `apply_coords_torch` function instead of `apply_coords` (which is what the error message is referencing) and similarly `predict_torch` instead of `predict` when using batched inputs.

If the batched inputs still cause problems, one simple alternative is to just run the regular (non-batched) `predict` function multiple times with different prompts to get multiple distinct masks. For example:
```
masks_1, _, _ = predictor.predict(input_points_1, input_labels_1)
masks_2, _, _ = predictor.predict(input_points_2, input_labels_2)
masks_3, _, _ = predictor.predict(input_points_3, input_labels_3)
etc.
```
The downside of doing it this way is that it will run slower than using batched inputs.

**Comment by ByungKwanLee** [comment]
https://github.com/ByungKwanLee/Full-Segment-Anything addresses the ciritical issues of SAM, which supports batch-input on the full-grid prompt (automatic mask generation) **with post-processing**: removing duplicated or small regions and holes, under **flexible input image size**
